### PR TITLE
Attach attachments to analysis on a worksheet when doing results

### PR DESCRIPTION
**Pull request by Lunga001**
instrument imports

## Description of the issue/feature this PR addresses

Linked issue: https://github.com/senaite/senaite.core/issues/852

## Current behavior before PR
Resutls updated but Attachment column on the worksheet view is empty
## Desired behavior after PR is merged
Attachment column on the worksheet view should have links to the attached files

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: bika/lims/exportimport/instruments/resultsimport.py [hunk 1]
```diff
@@ -662,21 +662,26 @@ def create_attachment(self, ws, infile):
             attachment.reindexObject()
         return attachment
 
-    def attach_attachment(self, analysis, attachment):
-        if attachment:
+    def attach_attachments(self, analysis, attachments):
+        # Sometime we get a list and sometimes an object of 1 attachment
+        import pdb; pdb.set_trace()
```
Comment (ramonski) [inline bika/lims/exportimport/instruments/resultsimport.py]: I think you should remove that line;)

**Comment by Lunga001** [comment]
I've tried to extend the `InstrumentsImportInterface.rst` test by including the the Worksheet on the test so that the results import file can be attached to the analysis but I get an error
```
'Traceback (most recent call last):\n  File "/home/lunga/instances/senaite-master-zeo/zeocluster/src/senaite.core/bika/lims/exportimport/instruments/generic/two_dimension.py", line 77, in Import
importer.process()\n  File "/home/lunga/instances/senaite-master-zeo/zeocluster/src/senaite.core/bika/lims/exportimport/instruments/resultsimport.py", line 572, in process
self.create_attachment(ws, infile)\n  File "/home/lunga/instances/senaite-master-zeo/zeocluster/src/senaite.core/bika/lims/exportimport/instruments/resultsimport.py", line 656, in create_attachment
 if attuid and infile:\n  File "/home/lunga/instances/senaite-master-zeo/buildout-cache/eggs/Products.Archetypes-1.9.17-py2.7.egg/Products/Archetypes/BaseObject.py", line 428, in update\n    self.Schema().updateAll(self, **kwargs)
File "/home/lunga/instances/senaite-master-zeo/buildout-cache/eggs/Products.Archetypes-1.9.17-py2.7.egg/Products/Archetypes/Schema/__init__.py", line 528, in updateAll
method(kwargs[name])
File "/home/lunga/instances/senaite-master-zeo/buildout-cache/eggs/Products.Archetypes-1.9.17-py2.7.egg/Products/Archetypes/ClassGen.py", line 76, in generatedMutator
return schema[name].set(self, value, **kw)
File "/home/lunga/instances/senaite-master-zeo/buildout-cache/eggs/plone.app.blob-1.5.19-py2.7.egg/plone/app/blob/field.py", line 247, in set
blobbable = IBlobbable(value)\nTypeError: (\'Could not adapt\', <zope.publisher.browser.FileUpload object at 0x7fd1ff926c10>, <InterfaceClass plone.app.blob.interfaces.IBlobbable>)\n'
```
This is because we are using `FileUpload` on the test. Any help to get past this error would be appreciated.

**Comment by xispa** [comment]
Hey @Lunga001 , @juangallostra did the same job here https://github.com/senaite/senaite.core/pull/877 and resolved two bugs that were related. Please, when you pick an issue, assign yourself, otherwise no one knows if there is someone already working on it. Thanks!

**Comment by xispa** [comment]
Superseded by https://github.com/senaite/senaite.core/pull/877

**Pull request closed without merging**